src/NuGet.Core/NuGet.Common/TempUtility.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+
+namespace NuGet.Common
+{
+    public class TempFile : IDisposable
```

Comment:
This needs clean up, copying classes from test to production code doesn't work. This shouldn't use any sub folders under /tmp for linux to avoid permission issues